lib/matplotlib/axes/_axes.py

```diff
@@ -418,6 +418,18 @@ def legend(self, *args, **kwargs):
     def _remove_legend(self, legend):
         self.legend_ = None
 
+    @docstring.dedent_interpd
+    def colorbar(self, mappable, cax=None, use_gridspec=True, **kw):
+        """
+        Create a colorbar for a ScalarMappable instance, *mappable*, with the
+        axes as the parent.
+
+        Documentation for the pyplot thin wrapper:
+        %(colorbar_doc)s
+        """
+        self.figure.colorbar(self, mappable, cax=None, ax=self,
+                             use_gridspec=True, **kw)
```

Comment:
I think it needs to be

    self.figure.colorbar(mappable, cax=cax, ax=self,
                             use_gridspec=use_gridspec, **kw)

and maybe you want to pop `ax` out of `kw` already within this method? Else (I think) it should raise an error if `ax` is given twice to as argument.

Reply:
Yep!  Sorry, should have marked work in progress, I havent' tested it , and it needs tests in `test_axes`